### PR TITLE
Zipkin exporter

### LINKED ISSUE #109: SpanData exporter implementation: Zipkin
The task is to create a SpanData exporter for exporting Zipkin

### PR DESCRIPTION
**Pull request by krnowak**
This adds a zipkin exporter. For converting span data from otel to zipkin, I tried to follow https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/exporter-zipkin.md

The tests are still missing.

~~This PR implements both `SpanSyncer` and `SpanBatcher` interfaces as a separate types. Not sure if this is a way to go, but the code was simple enough to have both.~~
Implements now only the `SpanBatcher` interface.

~~One place I was stumped was when I wanted to convert otel trace ID (which is an array of 16 bytes) into zipkin trace ID (which is a pair of high and low uint64s). What endianness to use?~~
Using big endian.

Fixes #109.

### CODE REVIEW

**Comment by codeboten** [comment]
It looks like it's using BigEndian according to some code in the repo:
https://github.com/openzipkin/zipkin-go/blob/62dc8b26c05e0e8b88eb7536eff92498e65bbfc3/proto/v2/encode_proto.go#L57-L58

**Comment by krnowak** [comment]
Pushed an update. Still no tests, but added an example with a README. I couldn't get the zipkin collector to show the spans, though. Maybe it's because of missing endpoint information - the collector replies with HTTP 202 (which might mean that the sent data is okish, but not yet processed?).

**Comment by jmacd** [comment]
This looks good. I think we should try to have _some_ sort of test, even if it's not fully validated against a real server at this time. Maybe just start an HTTP Server & check that you receive the expected data after a POST?

File: exporters/trace/zipkin/zipkin.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	}
+}
+
+func WithClient(client *http.Client) Option {
+	return func(opts *Options) {
+		opts.Client = client
+	}
+}
+
+func NewExporter(url string, os ...Option) (*Exporter, error) {
```
Comment (MrAlias) [inline exporters/trace/zipkin/zipkin.go]: Making `url` a [URL](https://golang.org/pkg/net/url/#URL) would provide a bit stronger guarantees around correctness. That way it isn't all the way to when the first Span is sent that it is figured out it is a malformed URL.

File: exporters/trace/zipkin/zipkin.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		// nothing to do
+		return
+	}
+	models := toZipkinSpanModels(batch)
+	body, err := json.Marshal(models)
+	if err != nil {
+		e.logf("failed to serialize zipkin models to JSON: %v", err)
+		return
+	}
+	e.logf("about to send a POST request to %s with body %s", e.url, body)
```
Comment (MrAlias) [inline exporters/trace/zipkin/zipkin.go]: This seems a bit verbose for normal operations.

**Comment by krnowak** [comment]
Pushed some changes before I saw the review. Mostly fixes and tests.

File: exporters/trace/zipkin/zipkin.go (hunk 2 above)
Comment (krnowak) [reply exporters/trace/zipkin/zipkin.go]: This is a no op by default, unless you configure the logger. And maybe it's lame, but I use that in tests too…

Comment (MrAlias) [reply exporters/trace/zipkin/zipkin.go]: Ah, didn't connect it was a no-op. SGTM

File: exporters/trace/zipkin/zipkin.go (hunk 1 above)
Comment (krnowak) [reply exporters/trace/zipkin/zipkin.go]: How about I validate that inside the `NewExporter` function? I haven't seen `url.URL` used anywhere as a parameter - we usually use strings.

Comment (krnowak) [reply exporters/trace/zipkin/zipkin.go]: Went with the validation inside the function for now.

**Comment by krnowak** [comment]
Updated again, added some docs.

**Pull request merged**